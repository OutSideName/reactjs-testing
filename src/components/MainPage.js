import React from "react";

import CardList from "./CardList";
import SearchBox from "./SearchBox";
import Scroll from "./Scroll";

class ComponentName extends React.Component {
  componentDidMount() {
    this.props.onRequestRobots();
  }

  filteredRobots = () => {
    let { searchField = "", robots = [] } = this.props;
    return robots.filter(robot =>
      robot.name.toLocaleLowerCase().includes(searchField)
    );
  };

  render() {
    const { onSearchChange, isPending } = this.props;

    if (isPending) {
      return <h1 className="tc">Loading...</h1>;
    }

    return (
      <div className="tc">
        <h1 className="f2">RoboFriends</h1>
        <SearchBox searchChange={onSearchChange} />
        <Scroll>
          <CardList robots={this.filteredRobots()} />
        </Scroll>
      </div>
    );
  }
}

export default ComponentName;
