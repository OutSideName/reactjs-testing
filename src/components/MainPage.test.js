import React from 'react';
import {shallow} from 'enzyme';
import MainPage from './MainPage';

let wrapper;

beforeEach(()=>{
  const mockProp = {
    onRequestRobots: jest.fn(),
    robots: [],
    searchField: '',
    isPending: false 
  }

  wrapper = shallow(<MainPage {...mockProp} />);
})

it("renders Mainpages with crashing", ()=> {
  expect(wrapper.instance()).toMatchSnapshot();
})

it("filter robots correctly", () => {
  expect(wrapper.instance().filteredRobots()).toEqual([]);
})