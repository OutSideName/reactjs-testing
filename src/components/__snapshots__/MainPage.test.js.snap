// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Mainpages with crashing 1`] = `
ComponentName {
  "context": Object {},
  "filteredRobots": [Function],
  "props": Object {
    "isPending": false,
    "onRequestRobots": [MockFunction] {
      "calls": Array [
        Array [],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    },
    "robots": Array [],
    "searchField": "",
  },
  "refs": Object {},
  "setState": [Function],
  "state": null,
  "updater": Updater {
    "_callbacks": Array [],
    "_renderer": ReactShallowRenderer {
      "_context": Object {},
      "_didScheduleRenderPhaseUpdate": false,
      "_dispatcher": Object {
        "readContext": [Function],
        "useCallback": [Function],
        "useContext": [Function],
        "useDebugValue": [Function],
        "useEffect": [Function],
        "useImperativeHandle": [Function],
        "useLayoutEffect": [Function],
        "useMemo": [Function],
        "useReducer": [Function],
        "useRef": [Function],
        "useState": [Function],
      },
      "_element": <ComponentName
        isPending={false}
        onRequestRobots={
          [MockFunction] {
            "calls": Array [
              Array [],
            ],
            "results": Array [
              Object {
                "type": "return",
                "value": undefined,
              },
            ],
          }
        }
        robots={Array []}
        searchField=""
      />,
      "_firstWorkInProgressHook": null,
      "_forcedUpdate": false,
      "_instance": [Circular],
      "_isReRender": false,
      "_newState": null,
      "_numberOfReRenders": 0,
      "_renderPhaseUpdates": null,
      "_rendered": <div
        className="tc"
      >
        <h1
          className="f2"
        >
          RoboFriends
        </h1>
        <SearchBox />
        <Scroll>
          <CardList
            robots={Array []}
          />
        </Scroll>
      </div>,
      "_rendering": false,
      "_updater": [Circular],
      "_workInProgressHook": null,
    },
  },
}
`;
